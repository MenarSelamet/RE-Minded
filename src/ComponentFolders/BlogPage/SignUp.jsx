import React, { useRef, useState, useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { useFormik } from 'formik';
import * as Yup from 'yup';
import { useNavigate } from 'react-router-dom';
import { addDoc, collection, getDocs } from 'firebase/firestore';
import emailjs from '@emailjs/browser';
import { db } from '../../Firebase';
import image4 from './Images/SubscribeArrow.svg';

const thanksProps =
  'Your email has been added to the mailing list successfully!';

function SignUp() {
  const emailList = collection(db, 'newsletter');
  const [emailArray, setArray] = useState([]);

   const form  = useRef();


  const navigate = useNavigate();

  const sendEmail = () => {

    emailjs.sendForm('service_lu1j8v5', 'template_ixxjvzh', form.current, 'da5Fk1wXTnepQQjPI')
      .then((result) => {
        // eslint-disable-next-line no-console
          console.log(result.text);
      }, (error) => {
        // eslint-disable-next-line no-console
          console.log(error.text);
      });
  };

  const formik = useFormik({
    initialValues: { email: '' },
    validationSchema: Yup.object({
      email: Yup.string()
        .email('Please enter a valid email')
        .required('Please enter an email'),
    }),
  });

  function handleFormSubmit() {
    if (!formik.values.email) {
      // eslint-disable-next-line no-alert
      alert('Please enter an email');
    } else if (formik.values.email) {
      if (
        formik.values.email
          .trim()
          .match(/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i)
      ) {
        const newArr = [];
        emailArray.map((data) => newArr.push(data.data.email));
        const result = newArr.includes(formik.values.email);
        // console.log(result)
        if (result === false) {
          sendEmail();
          addDoc(
            emailList,
            {
              email: formik.values.email,
            },
            navigate('/thankyou', { state: thanksProps })
          );
        } else {
          // eslint-disable-next-line no-alert
          alert('this email already exist');
        }
      } else {
        // eslint-disable-next-line no-alert
        alert('please enter a valid email');
      }
    }
  }

  useEffect(() => {
    const fetchData = async () => {
      const querySnapshot = await getDocs(emailList);
      const dataInfo = querySnapshot.docs.map((docu) => ({
        id: docu.id,
        data: docu.data(),
      }));
      setArray(dataInfo);
    };
    fetchData();
  }, [emailArray]);

  const { t } = useTranslation();

  return (
    <div className="lg:pl-48 pl-10">
      <h2 className=" mt-10 uppercase lg:text-2xl text-lg">
        {t('blog_singup.h1')}
      </h2>
      <p className=" mt-2 text-start md:text-sm text-xs uppercase">
      {t('blog_singup.t1')}
      </p>
      <div className="md:flex">
        <div className=" mt-7 flex justify-start mb-4 w-[300px] h-[50px] box-border rounded-md border-2 border-[#718096]">
        <form ref={form} className="w-full" onSubmit={formik.handleSubmit} >
              <div>
                <input
                name='email'
                  id="email"
                  onChange={formik.handleChange}
                  value={formik.values.email}
                  className=" w-full h-[46.5px] rounded-l-lg text-SubTexts text-black-800 placeholder:pl-2 placeholder:text-base"
                  type="text"
                  placeholder={t('blog_singup.button')}
                />
              </div>
            </form>
          <button
          onClick={handleFormSubmit}
            type="button"
            className="bg-Buttons rounded-r-lg w-1/4 h-auto"
          >
            <img
              className="w-[24px] h-[24px] top-[18px] left-[310px] items-center mx-auto"
              src={image4}
              alt="Arrow"
            />
          </button>
        </div>
      </div>
    </div>
  );
}

export default SignUp;
