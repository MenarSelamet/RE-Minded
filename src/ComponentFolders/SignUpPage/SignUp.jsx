import React, { useState } from 'react';
import Image from './Images/SofaImage.svg';
import lineImage from './Images/line.svg';
import FacebookLogo from './Images/FacebookLogo.svg';
import GoogleLogo from './Images/GoogleLogo.svg';

function SignUp() {
  const [signUpData, setSignUpData] = useState({
    userFirstName: '',
    userLastName: '',
    userEmail: '',
    userConfirmEmail: '',
    userPassword: '',
    userCondirmPassword: '',
    dayOfBirth: '',
    monthOfBirth: '',
    yearOfYear: '',
  });

  function handleOnClick(e) {
    const { value, name } = e.target;
    return setSignUpData((prevObj) => {
      return {
        ...prevObj,
        [name]: value,
      };
    });
  }

  return (
    <div className="h-screen flex justify-center content-center md:flex-wrap max-[767px]:flex-wrap gap-x-20">
      <img src={Image} alt="Sign up" />
      <div className="flex flex-col">
        <h2 className='text-5xl font-["Poppins"] font-normal mb-32 max-[767px]:mt-20 md:mt-20 max-[767px]:mb-10 md:mb-10'>
          SIGNUP NOW
        </h2>
        <form className="grid grid-rows-3 gap-4 shadow-2xl px-10 py-10 w-[555px] h-[493]">
          <div className="flex gap-x-7">
            <input
              type="text"
              placeholder="   First Name"
              name="userFirstName"
              onChange={(e) => handleOnClick(e)}
              className="h-14 w-56 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
              value={signUpData.userFirstName}
            />
            <input
              type="text"
              placeholder="   Last Name"
              name="userLastName"
              onChange={(e) => handleOnClick(e)}
              className="h-14 w-56 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
              value={signUpData.userLastName}
            />
          </div>
          <input
            type="text"
            placeholder="   Your Email"
            name="userEmail"
            onChange={(e) => handleOnClick(e)}
            className="h-14 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
            value={signUpData.userEmail}
          />
          <input
            type="text"
            placeholder="   Confirm email"
            name="userConfirmEmail"
            onChange={(e) => handleOnClick(e)}
            className="h-14 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
            value={signUpData.userConfirmEmail}
          />
          <div className="flex gap-x-7">
            <input
              type="password"
              placeholder="   Password"
              name="userPassword"
              onChange={(e) => handleOnClick(e)}
              className="h-14 w-56 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
              value={signUpData.userPassword}
            />
            <input
              type="password"
              placeholder="   Confirm password"
              name="userCondirmPassword"
              onChange={(e) => handleOnClick(e)}
              className="h-14 w-56 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
              value={signUpData.userCondirmPassword}
            />
          </div>
          <div className="flex items-center justify-between">
            <p className="mr-7 ml-7 font-light text-[#9DAFBD]">Birth Date</p>
            <input
              type="text"
              placeholder="  DD"
              name="dayOfBirth"
              onChange={(e) => handleOnClick(e)}
              className="h-14 w-12 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
              value={signUpData.dayOfBirth}
            />
            <input
              type="text"
              placeholder="  MM"
              name="monthOfBirth"
              onChange={(e) => handleOnClick(e)}
              className="h-14 w-12 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
              value={signUpData.monthOfBirth}
            />
            <input
              type="text"
              placeholder="  YYYY"
              name="yearOfYear"
              onChange={(e) => handleOnClick(e)}
              className="h-14 w-12 broder-solid border-2 border-[#D1DBE3] rounded-md w-36 placeholder-gray-300"
              value={signUpData.yearOfYear}
            />
          </div>
          <div className="flex justify-around py-3 gap-8">
            <button
              type="submit"
              className="bg-[#2DD3E3] font-medium text-2xl px-14 py-3 rounded-md shadow-[0px_7px_20px_rgba(0,0,0,0.2)]"
            >
              Login
            </button>
            <button
              type="submit"
              className="broder-solid border-2 border-[#2DD3E3] font-medium text-2xl px-14 rounded-md"
            >
              Signup
            </button>
          </div>
        </form>
        <div className="flex justify-around my-6">
          <img src={lineImage} alt="A line" />
          <p>Or</p>
          <img src={lineImage} alt="A line" />
        </div>
        <div className="flex justify-center my-6 gap-x-20">
          <img src={FacebookLogo} alt="Facebook logo" className='cursor-pointer'/>
          <img src={GoogleLogo} alt="Google logo" className='cursor-pointer'/>
        </div>
      </div>
    </div>
  );
}

export default SignUp;
