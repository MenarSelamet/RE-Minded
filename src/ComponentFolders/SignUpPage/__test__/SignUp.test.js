import renderer from 'react-test-renderer';
import SignUp from "../SignUp";

it('renders correctly', () => {
    const loginComponent = renderer
        .create(<SignUp />)
        .toJSON();
    expect(loginComponent).toMatchSnapshot();
});