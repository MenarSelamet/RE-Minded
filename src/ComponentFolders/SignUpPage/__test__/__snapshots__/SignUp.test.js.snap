// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="h-screen flex justify-center content-center md:flex-wrap max-[767px]:flex-wrap gap-x-20"
>
  <img
    alt="Sign up"
    src="SofaImage.svg"
  />
  <div
    className="flex flex-col"
  >
    <h2
      className="text-5xl font-[\\"Poppins\\"] font-normal mb-32 max-[767px]:mt-20 md:mt-20 max-[767px]:mb-10 md:mb-10"
    >
      SIGNUP NOW
    </h2>
    <form
      className="grid grid-rows-3 gap-4 shadow-2xl px-10 py-10 w-[555px] h-[493]"
    >
      <div
        className="flex gap-x-7"
      >
        <input
          className="h-14 w-56 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
          name="userFirstName"
          onChange={[Function]}
          placeholder="   First Name"
          type="text"
          value=""
        />
        <input
          className="h-14 w-56 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
          name="userLastName"
          onChange={[Function]}
          placeholder="   Last Name"
          type="text"
          value=""
        />
      </div>
      <input
        className="h-14 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
        name="userEmail"
        onChange={[Function]}
        placeholder="   Your Email"
        type="text"
        value=""
      />
      <input
        className="h-14 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
        name="userConfirmEmail"
        onChange={[Function]}
        placeholder="   Confirm email"
        type="text"
        value=""
      />
      <div
        className="flex gap-x-7"
      >
        <input
          className="h-14 w-56 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
          name="userPassword"
          onChange={[Function]}
          placeholder="   Password"
          type="password"
          value=""
        />
        <input
          className="h-14 w-56 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
          name="userCondirmPassword"
          onChange={[Function]}
          placeholder="   Confirm password"
          type="password"
          value=""
        />
      </div>
      <div
        className="flex items-center justify-between"
      >
        <p
          className="mr-7 ml-7 font-light text-[#9DAFBD]"
        >
          Birth Date
        </p>
        <input
          className="h-14 w-12 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
          name="dayOfBirth"
          onChange={[Function]}
          placeholder="  DD"
          type="text"
          value=""
        />
        <input
          className="h-14 w-12 broder-solid border-2 border-[#D1DBE3] rounded-md placeholder-gray-300"
          name="monthOfBirth"
          onChange={[Function]}
          placeholder="  MM"
          type="text"
          value=""
        />
        <input
          className="h-14 w-12 broder-solid border-2 border-[#D1DBE3] rounded-md w-36 placeholder-gray-300"
          name="yearOfYear"
          onChange={[Function]}
          placeholder="  YYYY"
          type="text"
          value=""
        />
      </div>
      <div
        className="flex justify-around py-3 gap-8"
      >
        <button
          className="bg-[#2DD3E3] font-medium text-2xl px-14 py-3 rounded-md shadow-[0px_7px_20px_rgba(0,0,0,0.2)]"
          type="submit"
        >
          Login
        </button>
        <button
          className="broder-solid border-2 border-[#2DD3E3] font-medium text-2xl px-14 rounded-md"
          type="submit"
        >
          Signup
        </button>
      </div>
    </form>
    <div
      className="flex justify-around my-6"
    >
      <img
        alt="A line"
        src="line.svg"
      />
      <p>
        Or
      </p>
      <img
        alt="A line"
        src="line.svg"
      />
    </div>
    <div
      className="flex justify-center my-6 gap-x-20"
    >
      <img
        alt="Facebook logo"
        className="cursor-pointer"
        src="FacebookLogo.svg"
      />
      <img
        alt="Google logo"
        className="cursor-pointer"
        src="GoogleLogo.svg"
      />
    </div>
  </div>
</div>
`;
